sdks/python/apache_beam/runners/portability/fn_api_runner.py

```diff
@@ -1304,17 +1351,25 @@ def query(self, filter=None):
 
 
 class RunnerResult(runner.PipelineResult):
-  def __init__(self, state, metrics_by_stage):
+  def __init__(self, state, metrics_by_stage, monitoring_infos_by_stage):
     super(RunnerResult, self).__init__(state)
     self._metrics_by_stage = metrics_by_stage
+    self._monitoring_infos_by_stage = monitoring_infos_by_stage
     self._user_metrics = None
 
   def wait_until_finish(self, duration=None):
     return self._state
 
-  def metrics(self):
+  def metrics(self, use_monitoring_infos=False, user_metrics_only=True):
```

Comment:
This type of change, is this reasonable? Adding these extra options on pulling out ht metrrics. This is currently the only way we have to test metrics, and I added the user_metrics_only, to allow testing the system metrics.

Note: We can update this code to only use use_monitoring_infos and delete all the legacy metrics once we are happy with this.  But I have made use of Metrics/MonitoringInfos optional for now

Reply:
This is a fine change, however I will prefer it to remove the legacy stuff and change to new metrics at the same time. Is there a reason to keep both for now?

Reply:
Based on the few tests I have done, I have matched the existing functionality. So we could go ahead and delete the old code if you would like. I just thought of this as a precautionary thing to do before flipping it on.

I think its okay to rename to RunnerResult.monitoring_infos.